Automattic/wp-calypso
# Tailored Flows: remove completingPurchase step and set up the site /start

#### Proposed Changes

* This creates a new function called `setupSiteAfterCreation`, which is a new version of `useSetupOnboardingSite` hook. It's not React-y and simply returns a promise.
* Then, in turn, call that function after creating the site to update the logo, pattern, title, tagline, and set the homepage.
* It removes the `completingPurchase` step and after checkout lands in the `launchpad` for LiB.

#### Testing Instructions
Since this affects normal and tailored flows. Please use the following table to test, and make the row you pick as done. 

<table>
<tr>
	<th>Flow
	<th>URL
	<th>Success criteria
	<th>Done
<tr>
	<td>Non tailored start - Free plan</td>
	<td>https://container-festive-haibt.calypso.live/start</td>
	<td>The site is created and all works as expected</td>
	<td></td>
<tr>
	<td>Non-tailored start - Paid plan</td>
	<td>https://container-festive-haibt.calypso.live/start</td>
    <td>The site is created, **and you're taken to checkout**</td>
	<td></td>
<tr>
	<td>Tailored flow - LiB - Free plan</td>
	<td>https://container-festive-haibt.calypso.live/setup?flow=link-in-bio</td>
    <td>The site must have the logo, pattern, title, and tagline set</td>
	<td></td>
<tr>
	<td>Tailored flow - LiB - Paid plan</td>
	<td>https://container-festive-haibt.calypso.live/setup?flow=link-in-bio</td>
    <td>The site must have the logo, title, and tagline set</td>
	<td>
<tr>
	<td>Tailored flow - Newsletter - Free plan</td>
	<td>https://container-festive-haibt.calypso.live/setup?flow=newsletter</td>
    <td>The site must have the logo, title, and tagline set</td>
	<td></td>
</table>

Fixes: https://github.com/Automattic/wp-calypso/issues/67367
